http-servers/src/main/java/com/palantir/remoting/http/server/WebPreconditions.java

```diff
@@ -0,0 +1,142 @@
+ * includes other useful web-oriented checks like {@link #checkNotEmpty(String)} and excludes less relevant ones (like
+ * {@link Preconditions#checkState(boolean)}.
+ */
+public final class WebPreconditions {
+
+    private WebPreconditions() {}
+
+    /**
+     * See {@link Preconditions#checkArgument(boolean, Object)}.
+     *
+     * @throws BadRequestException
+     *         when <code>expression</code> is false
+     */
+    public static void checkArgument(boolean expression, @Nullable Object message) {
+        if (!expression) {
+            throw new BadRequestException(String.valueOf(message));
+        }
+    }
+
+    /**
+     * See {@link Preconditions#checkArgument(boolean, String, Object...)}.
+     *
+     * @throws BadRequestException
+     *         when <code>expression</code> is false
+     */
+    public static void checkArgument(boolean expression, @Nullable String messageTemplate,
+            @Nullable Object... messageArgs) {
+        if (!expression) {
+            throw new BadRequestException(format(messageTemplate, messageArgs));
+        }
+    }
+
+    /**
+     * See {@link Preconditions#checkArgument(boolean, Object)} and {@link Strings#isNullOrEmpty(String)}.
+     *
+     * @throws BadRequestException
+     *         when <code>reference</code> is null or empty
+     */
+    public static String checkNotEmpty(String string, @Nullable Object message) {
+        checkArgument(!Strings.isNullOrEmpty(string), message);
+
+        return string;
+    }
+
+    /**
+     * See {@link Preconditions#checkArgument(boolean, String, Object...)} and {@link Strings#isNullOrEmpty(String)}.
+     *
+     * @throws BadRequestException
+     *         when <code>reference</code> is null or empty
+     */
+    public static String checkNotEmpty(String string, @Nullable String messageTemplate,
+            @Nullable Object... messageArgs) {
+        checkArgument(!Strings.isNullOrEmpty(string), messageTemplate, messageArgs);
+
+        return string;
+    }
+
+    /**
+     * See {@link Preconditions#checkNotNull(Object, Object)}.
+     *
+     * @throws BadRequestException
+     *         when <code>reference</code> is null
+     */
+    public static <T> T checkNotNull(T reference, @Nullable Object message) {
+        checkArgument(reference != null, message);
+
+        return reference;
+    }
+
+    /**
+     * See {@link Preconditions#checkNotNull(Object, Object)}.
+     *
+     * @throws BadRequestException
+     *         when <code>reference</code> is null
+     */
+    public static <T> T checkNotNull(T reference, @Nullable String messageTemplate, @Nullable Object... messageArgs) {
+        checkArgument(reference != null, messageTemplate, messageArgs);
+
+        return reference;
+    }
+
+    /*
+     * Taken almost verbatim from Guava's Preconditions#format(String, Object...) modulo some variable name
+     * tweaks to make checkstyle happy.
+     */
```

Comment:
cool, added the original copyright with mentions of the changes made in the file.